File: solutions/2026-03-04/solution.java
// Longest Common Prefix
// Difficulty: Easy
// URL: https://leetcode.com/problems/longest-common-prefix/

class Solution {
  public String longestCommonPrefix(String[] strs) {
    if (strs == null || strs.length == 0) return "";
    String minStr = strs[0];
    String maxStr = strs[0];
    for (String str : strs) {
      if (str.length() < minStr.length()) minStr = str;
      if (str.length() > maxStr.length()) maxStr = str;
    }
    for (int i = 0; i < minStr.length(); i++) {
      if (minStr.charAt(i) != maxStr.charAt(i)) return minStr.substring(0, i);
    }
    return minStr;
  }
}

File: solutions/2026-03-04/solution.py
# Longest Common Prefix
# Difficulty: Easy
# URL: https://leetcode.com/problems/longest-common-prefix/

def longestCommonPrefix(strs): 
  if not strs: return ""
  s1 = min(strs)
  s2 = max(strs)
  for i, (c1, c2) in enumerate(zip(s1, s2)):
    if c1 != c2: return s1[:i]
  return s1